modern/src/common/util/formatter.js

```diff
@@ -64,6 +65,8 @@ export const formatAltitude = (value, unit, t) => `${altitudeFromMeters(value, u
 
 export const formatSpeed = (value, unit, t) => `${speedFromKnots(value, unit).toFixed(2)} ${speedUnitString(unit, t)}`;
 
+export const formatSpeedKph = (value, unit, t) => `${speedFromKph(value, unit).toFixed(2)} ${speedUnitString(unit, t)}`;
```

Comment:
Same here. You should just convert the value and call `formatSpeed` instead of duplicating the code.

Reply:
Truly I don't like that ide either.. all speed values should be treated same, don't think it shoud be different speed types at all.. and why speed value was chosen to be in knots? And only obd speed in km/h? I would like to stickto idea that all values are stored in same format.. also in traccar repository i find couple obd_speed storied as knot... other's stored asis..

Reply:
I think you misunderstood something. I'm not suggesting that display units are limited. I'm telling you that you can reuse the existing code instead of duplicating it.

Reply:
@tananaev will it be better now?

Reply:
No. You removed what you shouldn't remove and didn't remove what you should.

Reply:
Sorry i am giving up :D something i didn't understand when.. 
So i just need to convert saved kmh value to knots, and then use same function to convert back to kmh?

Reply:
You just need to convert value to knots and pass it to the existing function.

Reply:
Yes, but it will add error convert and round operations.. saved value of 100km/h => (56.996) 56.0 kn=>  (100.008) 100.01km/h 

So why's my solution is so bad?

Reply:
The rounding shouldn't be a problem. We pass the value with very high accuracy, so rounded value should always be the same as the original.

Your solution is bad because it adds code that we don't need.

Reply:
Ok @tananaev made changes as you suggested, if it's ok please confirm PR.